compiler/fir/resolve/src/org/jetbrains/kotlin/fir/resolve/transformers/body/resolve/BodyResolveContext.kt

```diff
@@ -486,6 +486,11 @@ class BodyResolveContext(
         return withTowerDataCleanup {
             addLocalScope(FirLocalScope())
             if (function is FirSimpleFunction) {
+                // Make all value parameters available in the local scope so that even one parameter that refers to another parameter,
+                // which may not be initialized yet, could be resolved. We will detect and report such case in checkers.
+                for (parameter in function.valueParameters) {
+                    storeVariable(parameter)
```

Comment:
Should we try removing the `storeVariable` calls in `withValueParameter` if the function in question is a simple function or constructor? Otherwise, we are storing redunant variables in the context right?

Reply:
Yeah, I thought of that too, but then decided not to touch that because value parameter is also used in catch clause. Most of all, I'm not sure body resolver knows the context of transformation?

Anyway, info stored in the local scope is a _mapping_ from `Name` to variable symbol, so technically not redundant, but it will just override the mapping with the same name and same parameter.